src/lightning/pytorch/strategies/fsdp.py

```diff
@@ -637,3 +641,8 @@ def load_checkpoint(self, checkpoint_path: _PATH) -> Dict[str, Any]:
             f"The path {str(path)!r} does not point to a valid checkpoint. Make sure the path points to either a"
             " directory with FSDP checkpoint shards, or a single file with a full checkpoint."
         )
+
+
+def _move_torchmetrics_to_device(module: torch.nn.Module, device: torch.device) -> None:
+    for metric in (m for m in module.modules() if isinstance(m, Metric)):
+        metric.to(device)
```

Comment:
I could also generalize this to all modules that don't have parameters. Reviewers please let me know what you think.

Reply:
I think it should unless you can find proof in the FSDP source code that this is a bad idea.

DDP does move them so it's intuitive that FSDP would too.

Reply:
I changed my mind, I'd like to do it only for Metrics for now. The reason being that it gets too complicated, because `.to()` is recursive and applies to submodules, but that needs to be avoided since FSDP modules could be nested under modules that don't have params. I'd like to keep the fix focused here for the moment.

I also opened https://github.com/pytorch/pytorch/issues/113113 FYI